src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/CcdCaseUpdater.java

```diff
@@ -138,62 +148,105 @@ public ProcessResult updateCase(
             ClientServiceErrorResponse errorResponse = serviceResponseParser.parseResponseBody(exception);
             return new ProcessResult(errorResponse.warnings, errorResponse.errors);
         } catch (Conflict exception) {
-            String msg = format(
-                "Failed to update case for %s service with case Id %s based on exception record %s "
-                    + "because it has been updated in the meantime",
-                configItem.getService(),
-                existingCaseId,
-                exceptionRecord.id
-            );
+            String msg = getErrorMessage(configItem.getService(), existingCaseId, exceptionRecord.id)
+                + " because it has been updated in the meantime";
             log.error(msg);
             ClientServiceErrorResponse errorResponse = new ClientServiceErrorResponse(asList(msg), emptyList());
             return new ProcessResult(errorResponse.warnings, errorResponse.errors);
-        } catch (Exception exception) {
+        } catch (FeignException exception) {
+            log.error(
+                getErrorMessage(configItem.getService(), existingCaseId, exceptionRecord.id)
+                    + "Service response: {}",
+                configItem.getService(),
+                existingCaseId,
+                exceptionRecord.id,
+                exception.contentUTF8(),
+                exception
+            );
+
             throw new CallbackException(
                 format(
-                    "Failed to update case for %s service with case Id %s based on exception record %s",
-                    configItem.getService(),
-                    existingCaseId,
-                    exceptionRecord.id
+                    "%s. Service response: %s",
+                    getErrorMessage(configItem.getService(), existingCaseId, exceptionRecord.id),
+                    exception.contentUTF8()
                 ),
                 exception
             );
+        } catch (Exception exception) {
+            throw new CallbackException(
+                getErrorMessage(configItem.getService(), existingCaseId, exceptionRecord.id),
+                exception
+            );
         }
     }
 
-    private void updateCaseInCcd(
+    private String getErrorMessage(String service, String existingCaseId, String exceptionRecordId) {
+        return format(
+            "Failed to update case for %s service "
+                + "with case Id %s "
+                + "based on exception record %s",
+            service,
+            existingCaseId,
+            exceptionRecordId
+        );
+    }
+
+    private Either<List<String>, Boolean> updateCaseInCcd(
```

Comment:
Looks like we could just use optional error here since right value is obsolete and left value - always a single string. So signature becomes just `Optional<String>` here

Reply:
docblock should explain why such (one or another) signature is here - what does it mean

Reply:
Fixed this. Just tried to follow the usual approach when Either is used to distinguish between successful and failed execution.